File: Test/Test/GuessingGame.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Test
{
    class GuessingGame
    {
        public int maxGuesses = 3;

        public void PlayGame()
        {
            Random numberGen = new Random();
            int number = numberGen.Next(0, 5);

            Console.WriteLine("You have 3 trys to guess a number between 1 and 5!");

            int guesses = 0;
            while (guesses != maxGuesses)
            {
                int guess = Convert.ToInt32(Console.ReadLine());

                if (guess == number)
                {
                    Console.WriteLine("You have guessed correctly, well done!");
                    return;
                }
                else
                {
                    Console.WriteLine("You have guessed incorrectly, guess again");
                    guesses++;
                }
            }

            Console.WriteLine("Game Over! You have ran out of guesses");
        }

    }
}

File: Test++/Test++/GuessingGame.cpp
#include "GuessingGame.h"
#include <iostream>

using namespace std;

GuessingGame::GuessingGame(int max)
{
	maxGuesses = max;
}

void GuessingGame::PlayGame()
{
	srand((unsigned)time(0));

	int numbGuesses = 0;
	int guess;
	int number = 1 +(rand() % 5);

	cout << "You have " << maxGuesses << " guesses to find the correct number." << endl;

	while (numbGuesses != maxGuesses)
	{
		cin >> guess;

		if (guess == number)
		{
			cout << "You have guessed correctly, well done!\n" << endl;
			return;
		}
		else
		{
			cout << "You have guessed incorrectly, try again." << endl;
			numbGuesses++;
		}
	}

	cout << "GameOver, the hidden number was " << number << "\n" << endl;
}
